nautobot/extras/tests/test_api.py

```diff
@@ -2627,6 +2657,191 @@ def test_populate_relationship_associations_on_site_create(self):
+                    '["source"]["objects"].',
+                ]
+            }
+        ]
+        self.assertHttpStatus(response, 400)
+        self.assertEqual(expected_error_json, response.json())
+        # Check that no VLAN was created:
+        self.assertEqual(VLAN.objects.count(), existing_vlans_count)
+
+        # Try creating a vlan when we haven't specified any device data in the list of required objects
+        device1 = create_test_device("Device 1")
+        device2 = create_test_device("Device 2")
+        response = self.post_to_endpoint(VLAN, create_data)
+        expected_error_json = [
+            {
+                "vlans-devices": [
+                    'You must specify at least one device [<device.id>, ...] in relationships["vlans-devices"]'
+                    '["source"]["objects"].'
+                ]
+            },
+        ]
+        self.assertHttpStatus(response, 400)
+        self.assertEqual(expected_error_json, response.json())
+        # Check that no VLAN was created:
+        self.assertEqual(VLAN.objects.count(), existing_vlans_count)
+
+        # Try creating a vlan when all two devices are provided for the m2m relationship
+        create_data["relationships"] = {
+            "vlans-devices": {
+                "source": {
+                    "objects": [
+                        str(device1.pk),
+                        str(device2.pk),
+                    ]
+                }
+            }
+        }
+        response = self.post_to_endpoint(VLAN, create_data)
+        self.assertHttpStatus(response, 201)
+        # Check that a VLAN was created:
+        self.assertEqual(VLAN.objects.count(), existing_vlans_count + 1)
+
+    def test_create_platform_with_required_device_o2m(self):
+        """
+        Test that we can't create a Platform without specifying a Device for this required relationship
+        1. Try creating a Platform when no Device exists
+        2. Try creating a Platform with no specified Device data in the list of required objects
+        3. Try creating a Platform when all required data is present
+        """
+        self.user.is_superuser = True
+        self.user.save()
+
+        existing_platforms_count = Platform.objects.count()
+
+        create_data = {
+            "name": "New Platform",
+            "status": "active",
+        }
+
+        # Try creating a platform when no device exists
+        response = self.post_to_endpoint(Platform, create_data)
+        expected_error_json = [
+            {
+                "platform-devices": [
+                    "Platforms require at least one device, but no devices exist yet. "
+                    "Create a device by posting to /api/dcim/devices/",
+                    'You must specify at least one device [<device.id>, ...] in relationships["platform-devices"]'
+                    '["destination"]["objects"].',
```

Comment:
This seems a bit confusing to me - I could read this as saying that you need to specify `relationships["platform-devices"]["destination"]['objects"]` when posting to `/api/dcim/devices/`?